src/olympia/addons/tests/test_views.py

```diff
@@ -2818,6 +2818,24 @@ def test_basic(self):
 
         assert {itm['id'] for itm in data['results']} == {addon.pk, addon2.pk}
 
+    def test_default_locale_fallback_still_works_for_translations(self):
+        addon = addon_factory(default_locale='pt-BR', name='foobar')
+        # Couple quick checks to make sure the add-on is in the right state
+        # before testing.
+        assert addon.default_locale == 'pt-BR'
+        assert addon.name.locale == 'pt-br'
+
+        self.refresh()
+
+        # Search in a different language than the one used for the name: we
+        # should fall back to default_locale and find the translation.
+        data = self.perform_search(self.url, {'q': 'foobar', 'lang': 'fr'})
+        assert data['results'][0]['name'] == 'foobar'
+
+        # Same deal in en-US.
+        data = self.perform_search(self.url, {'q': 'foobar', 'lang': 'en-US'})
+        assert data['results'][0]['name'] == 'foobar'
```

Comment:
The value of `default_locale` should definitively be tested here. I honestly also wonder if we should return the locale that string is actually in - probably doesn't matter much since we can't really act on it but might help debugging quite a bit.

Reply:
We can't test the value of `default_locale`, it's not returned by the autocomplete API. You're supposed to always pass `lang` when using it, and therefore you don't really care about what the `default_locale` is - you just want a string.

And it might be a little too late to return extra info about the string... Ideally instead of returning a string directly when `lang` is passed we should continue to send an object, but with only one key/value corresponding to the most relevant language for the request. But it's a little late to make such a backwards-incompatible change...